src/AvoDeduplicator.ts

```diff
@@ -0,0 +1,183 @@
+    checkInAvoFunctions: boolean
+  ) {
+    let result = false;
+
+    if (checkInAvoFunctions) {
+      for (const otherEventName in this.avoFunctionsEventsParams) {
+        if (this.avoFunctionsEventsParams.hasOwnProperty(otherEventName)) {
+          const otherParams = this.avoFunctionsEventsParams[otherEventName];
+          if (deepEquals(params, otherParams)) {
+            result = true;
+            break;
+          }
+        }
+      }
+    } else {
+      for (const otherEventName in this.manualEventsParams) {
+        if (this.manualEventsParams.hasOwnProperty(otherEventName)) {
+          const otherParams = this.manualEventsParams[otherEventName];
+          if (deepEquals(params, otherParams)) {
+            result = true;
+            break;
+          }
+        }
+      }
+    }
+
+    return result;
+  }
+
+  shouldRegisterSchemaFromManually(
+    eventName: string,
+    eventSchema: Array<{
+      propertyName: string;
+      propertyType: string;
+      children?: any;
+    }>
+  ): boolean {
+    this.clearOldEvents();
+
+    return !this.hasSameShapeInAvoFunctionsAs(eventName, eventSchema);
+  }
+
+  private hasSameShapeInAvoFunctionsAs(
+    eventName: string,
+    eventSchema: Array<{
+      propertyName: string;
+      propertyType: string;
+      children?: any;
+    }>
+  ): boolean {
+    let result = false;
+
+    for (const otherEventName in this.avoFunctionsEventsParams) {
+      if (this.avoFunctionsEventsParams.hasOwnProperty(otherEventName)) {
+        if (otherEventName === eventName) {
+          const otherSchema = AvoSchemaParser.extractSchema(
+            this.avoFunctionsEventsParams[eventName]
+          );
+          if (deepEquals(eventSchema, otherSchema)) {
+            result = true;
+            break;
+          }
+        }
+      }
+    }
+
+    if (result) {
+      delete this.avoFunctionsEventsParams[eventName];
+    }
+
+    return result;
+  }
+
+  private clearOldEvents() {
+    const now = Date.now();
+    const msToConsiderOld = 300;
+
+    for (const time in this.avoFunctionsEvents) {
+      if (this.avoFunctionsEvents.hasOwnProperty(time)) {
+        const timestamp = Number(time) || 0;
+        if (now - timestamp > msToConsiderOld) {
+          const eventName = this.avoFunctionsEvents[time];
+          delete this.avoFunctionsEvents[time];
+          delete this.avoFunctionsEventsParams[eventName];
+        }
+      }
+    }
+
+    for (const time in this.manualEvents) {
+      if (this.manualEvents.hasOwnProperty(time)) {
+        const timestamp = Number(time) || 0;
+        if (now - timestamp > msToConsiderOld) {
+          const eventName = this.manualEvents[time];
+          delete this.manualEvents[time];
+          delete this.manualEventsParams[eventName];
+        }
+      }
+    }
+  }
+
+  private _clearEvents() {
+    this.avoFunctionsEvents = {};
+    this.manualEvents = {};
+
+    this.avoFunctionsEventsParams = {};
+    this.manualEventsParams = {};
+  }
+}
```

Comment:
I feel like this would be a good file to write tests for 👀😬

Reply:
exactly :)